packages/zoe/zoe.js

```diff
@@ -9,31 +9,28 @@ import { isOfferSafeForAll } from './isOfferSafe';
 import { areRightsConserved } from './areRightsConserved';
 import { evalContractCode } from './evalContractCode';
 import { makeTables } from './state';
-import { makeSeatMint } from './seatMint';
-import { makeEscrowReceiptConfig } from './escrowReceiptConfig';
+import { makeInviteConfig } from './inviteConfig';
 
 /**
  * Create an instance of Zoe.
  *
  * @param additionalEndowments pure or pure-ish endowments to add to evaluator
  */
 const makeZoe = (additionalEndowments = {}) => {
-  // Zoe has two mints: a mint for invites and a mint for
-  // escrowReceipts. The invite mint can be used by a smart contract
-  // to create invites to take certain actions in the smart contract.
-  // An escrowReceipt is an ERTP payment that is proof of
-  // escrowing assets with Zoe.
-  const {
-    seatMint: inviteMint,
-    seatAssay: inviteAssay,
-    addUseObj: addInviteSeat,
-  } = makeSeatMint('zoeInvite');
-
-  const escrowReceiptMint = makeMint(
-    'zoeEscrowReceipts',
-    makeEscrowReceiptConfig,
-  );
-  const escrowReceiptAssay = escrowReceiptMint.getAssay();
+  const makeInviteMint = () => {
+    const handleToSeat = new WeakMap();
+    const inviteMint = makeMint('zoeInvite', makeInviteConfig);
+    return harden({
+      inviteAssay: inviteMint.getAssay(),
+      mintInvite: (handle, seat, units) => {
+        handleToSeat.set(handle, seat);
+        const purse = inviteMint.mint(units);
+        return purse.withdrawAll();
+      },
+      getInviteSeat: handle => handleToSeat.get(handle),
+    });
+  };
+  const { inviteAssay, mintInvite, getInviteSeat } = makeInviteMint();
```

Comment:
If I'm reading this right, `inviteAssay` is a function that always returns the same assay. Why not just bind it directly to the assay?

In fact, it looks like `makeInviteMint` is only used here, and the object it returns is only used to initialize these three variables. Why not just unpack the code of `makeInviteMint` into code that initializes these three variables directly?

Reply:
inviteAssay is bound directly to the assay: `inviteAssay: inviteMint.getAssay(),`

I wanted makeInviteMint to be a separate function for POLA. `handleToSeat` isn't accessible outside the function, and the only access is through `mintInvite` and `getInviteSeat`.

Reply:
I just pushed up a commit that exposes `handleToSeat` and unpacks the code. I still don't like exposed weakMaps, but it is simpler to understand what is happening here.

Reply:
I like the new code better than the old code here. But I appreciate the POLA concern. We can revisit that sometime if you like, but let's stick with your new code for now.

Reply:
Sounds good!